### ISSUE TITLE
mkdir: /lib/node_modules/.hooks: Permission denied

### ISSUE THREAD

**Issue by peterlobster** [nodenv/nodenv#107]
I had this pop up when I ran `nodenv install 9`, I don't know if this is normal

```bash
$ nodenv install 9
node-build: definition not found: 9

The following versions contain `9' in the name:
  0.1.19
  0.1.29
  0.1.90
  0.1.91
  0.1.92
  0.1.93
  0.1.94
  0.1.95
  0.1.96
  0.1.97
  0.1.98
  0.1.99
  0.4.9
  0.5.9
  0.6.9
  0.6.19
  0.7.9
  0.8.9
  0.8.19
  0.9.0
  0.9.1
  0.9.2
  0.9.3
  0.9.4
  0.9.5
  0.9.6
  0.9.7
  0.9.8
  0.9.9
  0.9.10
  0.9.11
  0.9.12
  0.10.9
  0.10.19
  0.10.29
  0.10.39
  0.11.9
  0.12.9
  4.9.0
  4.9.1
  5.9.0
  5.9.1
  6.9.0
  6.9.1
  6.9.2
  6.9.3
  6.9.4
  6.9.5
  7.9.0
  8.9.0
  8.9.1
  8.9.2
  8.9.3
  8.9.4
  9.0.0
  9.1.0
  9.2.0
  9.2.1
  9.3.0
  9.4.0
  9.5.0
  9.6.0
  9.6.1
  9.7.0
  9.7.1
  9.8.0
  9.9.0
  9.10.0
  9.10.1
  9.11.0
  9.11.1
  9.11.2

See all available versions with `nodenv install --list'.

If the version you need is missing, try upgrading node-build:

  brew update && brew upgrade node-build
nodenv: version `9' not installed
**mkdir: /lib/node_modules/.hooks: Permission denied**

```

**Comment by jasonkarns** [comment]
Node-build doesn't yet support installing using aliases, so you need to select an exact version number.

One of:
```
9.0.0
  9.1.0
  9.2.0
  9.2.1
  9.3.0
  9.4.0
  9.5.0
  9.6.0
  9.6.1
  9.7.0
  9.7.1
  9.8.0
  9.9.0
  9.10.0
  9.10.1
  9.11.0
  9.11.1
  9.11.2
```

The `**mkdir: /lib/node_modules/.hooks: Permission denied**` error is the result of a plugin that must be running some code after installation. _That_ part seems like a bug, since it should exit cleanly if the node-build install wasn't successful. Could you share the list of nodenv plugins you have installed?

**Comment by peterlobster** [comment]
@jasonkarns Thanks for your help! I have pretty much every plugin I think. I've used them all at one time or the other, so I like to keep them all installed. I've attached a screenshot of my Cakebrew.[https://ibb.co/hF4o6U](Screen Shot of CakeBrew)

**Comment by peterlobster** [comment]
@jasonkarns Any update on what may be cause this? and a solution?

**Comment by jasonkarns** [comment]
Looks like the bug is with nodenv-package-rehash. But the only "bug" here is the message: `**mkdir: /lib/node_modules/.hooks: Permission denied**`. 

The initial error message: 

```
node-build: definition not found: 9

The following versions contain `9' in the name:
...
```

is not a bug. So even if we fix the bug with nodenv-package-rehash, the only change to the behavior here is that the expected output would be:

```
$ nodenv install 9
node-build: definition not found: 9

The following versions contain `9' in the name:
  0.1.19
  0.1.29
  0.1.90
  0.1.91
  0.1.92
  0.1.93
  0.1.94
  0.1.95
  0.1.96
  0.1.97
  0.1.98
  0.1.99
  0.4.9
  0.5.9
  0.6.9
  0.6.19
  0.7.9
  0.8.9
  0.8.19
  0.9.0
  0.9.1
  0.9.2
  0.9.3
  0.9.4
  0.9.5
  0.9.6
  0.9.7
  0.9.8
  0.9.9
  0.9.10
  0.9.11
  0.9.12
  0.10.9
  0.10.19
  0.10.29
  0.10.39
  0.11.9
  0.12.9
  4.9.0
  4.9.1
  5.9.0
  5.9.1
  6.9.0
  6.9.1
  6.9.2
  6.9.3
  6.9.4
  6.9.5
  7.9.0
  8.9.0
  8.9.1
  8.9.2
  8.9.3
  8.9.4
  9.0.0
  9.1.0
  9.2.0
  9.2.1
  9.3.0
  9.4.0
  9.5.0
  9.6.0
  9.6.1
  9.7.0
  9.7.1
  9.8.0
  9.9.0
  9.10.0
  9.10.1
  9.11.0
  9.11.1
  9.11.2

See all available versions with `nodenv install --list'.

If the version you need is missing, try upgrading node-build:

  brew update && brew upgrade node-build
```

**Comment by jasonkarns** [comment]
Ah, looks like this issue was solved in v1.2.0 of nodenv-package-rehash. (You're using 1.0.2)

https://github.com/nodenv/nodenv-package-rehash/releases/tag/v1.2.0

I'd be sure you're running recent versions of these plugins. nodenv-package-rehash 1.2.0 was released in 2015.
